pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -341,72 +342,158 @@ func (p *AgentPoller) Stop() {
 // With libvirt 5.6.0 direct call to agent can be replaced with call to libvirt Domain.GetGuestInfo
 func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
 	for _, command := range commands {
-		// replace with direct call to libvirt function when 5.6.0 is available
-		cmdResult, err := con.QemuAgentCommand(`{"execute":"`+string(command)+`"}`, domainName)
-		if err != nil {
-			// skip the command on error, it is not vital
-			continue
-		}
-
 		// parse the json data and convert to domain api
 		// for libvirt 5.6.0 json conversion deprecated
 		switch command {
 		case GET_INTERFACES:
-			interfaces, err := parseInterfaces(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent interface %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_INTERFACES, con, domainName); err == nil {
+				agentStore.Store(GET_INTERFACES, convertToInterfaces(guestInfo))
 			}
-			agentStore.Store(GET_INTERFACES, interfaces)
 		case GET_OSINFO:
-			osInfo, err := parseGuestOSInfo(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent guestosinfo %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_OS, con, domainName); err == nil {
+				agentStore.Store(GET_OSINFO, convertToOSInfo(guestInfo))
 			}
-			agentStore.Store(GET_OSINFO, osInfo)
 		case GET_HOSTNAME:
-			hostname, err := parseHostname(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent hostname %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_HOSTNAME, con, domainName); err == nil {
+				agentStore.Store(GET_HOSTNAME, guestInfo.Hostname)
 			}
-			agentStore.Store(GET_HOSTNAME, hostname)
 		case GET_TIMEZONE:
-			timezone, err := parseTimezone(cmdResult)
-			if err != nil {
-				log.Log.Errorf("Cannot parse guest agent timezone %s", err.Error())
-				continue
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_TIMEZONE, con, domainName); err == nil {
+				agentStore.Store(GET_TIMEZONE, convertToTimezone(guestInfo))
 			}
-			agentStore.Store(GET_TIMEZONE, timezone)
 		case GET_USERS:
-			users, err := parseUsers(cmdResult)
+			if guestInfo, err := getGuestInfo(libvirt.DOMAIN_GUEST_INFO_USERS, con, domainName); err == nil {
+				agentStore.Store(GET_USERS, convertToUsers(guestInfo))
+			}
+		case GET_FSFREEZE_STATUS:
+			// TODO: Replace with direct call to libvirt function
+			cmdResult, err := con.QemuAgentCommand(`{"execute":"`+string(command)+`"}`, domainName)
```

Comment:
This `guest-fsfreeze-freeze` is not exposed in libvirt API, so either we should keep using it this way (or check if to remove it from the project if it's not used anymore.).

There is an issue about it: [Add fsfreeze-status information to virDomainGetGuestInfo](https://gitlab.com/libvirt/libvirt/-/issues/706).